### ISSUE TITLE
False Positive Style/RedundantSelf when overriding accessor locally

### ISSUE THREAD

**Issue by envygeeks** [rubocop-hq/rubocop#5617]
## Expected behavior

No warning.

## Actual behavior

TLDR: Ruby predefines a reference for `var1` when doing `var1 = "" if self.var1` so `var1 = "" if var1` will evaluate to `nil` when it comes down to parse/exec time, and since it's a complex accessor, using `@var1` is unacceptable.

Rubocop throws a warning `Style/RedundantSelf` when using complex accessors, with locals of the same name, claiming that the `self` is redundant, when infact it's not, you cannot use `@var1` because `self.var1` is complex, and you cannot use `var1` because it's been set locally, and Ruby will predefine a reference to `var1`. The same way it will predefine references later, unless you enter into a separate context, then you have to explicitly define it, this is getting pedantic though.

## Steps to reproduce the problem

```ruby
class MyClass
  attr_writer :var1
  attr_writer :var2

  def initialize(var1: nil, var2: nil)
    @var1 = var1
    @var2 = var2
  end

  def var1
    @var1 ||= begin
      # Complex
      # NilClass
    end
  end

  def var2
    @var2 ||= begin
      # Complex
      # NilClass
    end
  end

  def inspect
    var1 = "blah #{self.var1}" if self.var1
    var2 = "blah #{self.var2}" if self.var2
    "#<#{self.class.name}#{var1}#{var2}"
  end
end
```
## RuboCop version

```
0.52.1 (using Parser 2.4.0.2, running on ruby 2.5.0 x86_64-darwin17)
```

**Comment by envygeeks** [comment]
I don't actually know if this is still a problem, or if it is... I've not noticed it anymore, this can probably be closed, I'll comment or open another issue if it arises again, but it's not been a problem since that day so I assume this is pretty stale and doesn't really matter tbqf.

@bbatsov

**Comment by Drenmi** [comment]
Thanks @envygeeks! 🙇
